### ISSUE TITLE
Find a better way to handle Flannel CNI

### ISSUE THREAD

**Issue by xmudrii** [kubernetes-sigs/cluster-api-provider-digitalocean#103]
**User story**

We're deploying Flannel CNI in [the Provider Components manifest, as part of the bootstrap script](https://github.com/kubermatic/cluster-api-provider-digitalocean/blob/908ea6ee6d61e6d1a05e73454b6835d022d819c5/clusterctl/examples/digitalocean/provider-components.yaml.template#L268-L272). This is a bad practice, [as CNI should not be handled by Cluster API, as per feature model](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/45b2ea80f5b5e51d4bc6a041a554e080bea13e70/docs/proposal/features.md#out-of-scope).

At this point, it isn't a problem as the bootstrap script and the provider components manifest are easily customizable, but may be a problem if we move bootstrap scripts out of the repository or when we implement custom images.

Instead, we should use the cluster add-ons if possible or find a better way to handle CNIs.

Using the cluster add-ons could be a great fit, but cluster add-ons are deployed only once cluster is ready, however cluster will never become ready if CNI is not deployed.

This is also important so operators can choose CNI other than Flannel.

**Acceptance criteria**

- [ ] Flannel CNI can be deployed as Cluster Add-on or there is a better way to handle CNIs.
